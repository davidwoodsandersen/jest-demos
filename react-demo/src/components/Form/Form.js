import React from 'react';
import './Form.scss';
import Button from '../Button/Button';

function Form(props) {
  var input;
  const handleSubmit = (e) => {
    e.preventDefault();
    props.addTodo(input);
  };

  return (
    <form className="Form" onSubmit={handleSubmit}>
      <input type="text" ref={(i) => input = i} />
      <Button />
    </form>
  );
}

export default Form;
