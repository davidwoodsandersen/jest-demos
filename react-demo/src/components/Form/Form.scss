.Form {
  input[type="text"] {
    width: 100%;
    padding: 8px;
    font-size: 16px;
    margin-bottom: 5px;
  }
}
