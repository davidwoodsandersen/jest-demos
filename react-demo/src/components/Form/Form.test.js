import React from 'react';
import ReactDOM from 'react-dom';
import Form from './Form';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Form />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
  const instance = renderer.create(<Form />).toJSON();
  expect(instance).toMatchSnapshot();
});
