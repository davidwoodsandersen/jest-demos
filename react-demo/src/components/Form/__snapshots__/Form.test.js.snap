// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  className="Form"
  onSubmit={[Function]}
>
  <input
    type="text"
  />
  <button
    className="Button"
  >
    Add Todo
  </button>
</form>
`;
